src/native/corehost/hostmisc/pal.windows.cpp

```diff
@@ -898,8 +898,12 @@ bool pal::realpath(pal::string_t* path, bool skip_error_logging)
                 }
             }
 
-            // Remove the \\?\ prefix, unless it is necessary or was already there
-            if (LongFile::IsExtended(str) && !LongFile::IsExtended(*path) &&
+            // Remove the UNC prefix (\\?\UNC\) or extended prefix (\\?\) unless it is necessary or was already there
+            if (LongFile::IsUNCExtended(str) && !LongFile::IsUNCExtended(*path) && !LongFile::ShouldNormalize(*path))
+            {
+                str = *path;
```

Comment:
Ah, actually this is tricky because after UNC is added, it's not really a prefix anymore. That is, if you have

`\\a\b\c`

then the UNC path is

`\\?\UNC\a\b\c`

So the proper replacement would be `\\?\UNC\ -> \\`. I'm not sure whether that's right.

Reply:
I think I've concluded this replacement is safe in all cases we care about (< MAX_PATH).